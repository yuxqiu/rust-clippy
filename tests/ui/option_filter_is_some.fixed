//@aux-build:option_helpers.rs
#![warn(clippy::option_filter_is_some)]

#[macro_use]
extern crate option_helpers;

#[rustfmt::skip]
fn option_methods() {
    let opt = Some(1);

    // Check for `option.filter(_).is_some()` use.
    // Single line case.
    let _ = opt.is_some_and(|x| x > 1);
    // Multi-line cases.
    let _ = opt.is_some_and(|x| {
        x > 1
    }
    );

    let y = 2;
    let _ = opt.is_some_and(|_| y > 1);

    let _ = opt.is_some_and(|x| (&x).eq(&5));

    let _ = opt.is_some_and(|x| x > 1);

    // won't fix because the argument for the closure is type annotated
    let _ = opt.filter(|&x: &i32| x > 1).is_some();

    // won't fix because
    // - the closure is bounded to a variable
    // - the argument for the closure is type annotated
    let predicate = |x: &i32| *x > 1;
    let _ = opt.filter(predicate).is_some();

    fn check_num(num: &i32) -> bool {
        *num > 1
    }

    // won't fix because `check_num` is not a closure
    let _ = opt.filter(check_num).is_some();

    // won't fix because of macro
    let _ = opt_filter!(opt, |x| *x == 1).is_some();
}
