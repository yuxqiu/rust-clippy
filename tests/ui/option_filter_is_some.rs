//@aux-build:option_helpers.rs
#![warn(clippy::option_filter_is_some)]

#[macro_use]
extern crate option_helpers;

#[rustfmt::skip]
fn option_methods() {
    let opt = Some(1);

    // Check for `option.filter(_).is_some()` use.
    // Single line case.
    let _ = opt.filter(|x| *x > 1)
        // Should lint even though this call is on a separate line.
        .is_some();
    // Multi-line cases.
    let _ = opt.filter(|x| {
        *x > 1
    }
    ).is_some();

    let y = 2;
    let _ = opt.filter(|_| y > 1).is_some();

    let _ = opt.filter(|x| x.eq(&5)).is_some();

    let _ = opt.filter(|&x| x > 1).is_some();

    // won't fix because the argument for the closure is type annotated
    let _ = opt.filter(|&x: &i32| x > 1).is_some();

    // won't fix because
    // - the closure is bounded to a variable
    // - the argument for the closure is type annotated
    let predicate = |x: &i32| *x > 1;
    let _ = opt.filter(predicate).is_some();

    fn check_num(num: &i32) -> bool {
        *num > 1
    }

    // won't fix because `check_num` is not a closure
    let _ = opt.filter(check_num).is_some();

    // won't fix because of macro
    let _ = opt_filter!(opt, |x| *x == 1).is_some();
}
