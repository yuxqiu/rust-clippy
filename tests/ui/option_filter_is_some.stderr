error: called `filter(<f>).is_some()` on an `Option` value
  --> $DIR/option_filter_is_some.rs:13:17
   |
LL |       let _ = opt.filter(|x| *x > 1)
   |  _________________^
LL | |         // Should lint even though this call is on a separate line.
LL | |         .is_some();
   | |__________________^
   |
   = note: `-D clippy::option-filter-is-some` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::option_filter_is_some)]`
help: use `is_some_and(<f>)` instead
   |
LL -     let _ = opt.filter(|x| *x > 1)
LL +     let _ = opt.is_some_and(|x| x > 1);
   |

error: called `filter(<f>).is_some()` on an `Option` value
  --> $DIR/option_filter_is_some.rs:17:17
   |
LL |       let _ = opt.filter(|x| {
   |  _________________^
LL | |         *x > 1
LL | |     }
LL | |     ).is_some();
   | |_______________^
   |
help: use `is_some_and(<f>)` instead
   |
LL ~     let _ = opt.is_some_and(|x| {
LL ~         x > 1
LL |     }
LL ~     );
   |

error: called `filter(<f>).is_some()` on an `Option` value
  --> $DIR/option_filter_is_some.rs:23:17
   |
LL |     let _ = opt.filter(|_| y > 1).is_some();
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: use: `is_some_and(|_| y > 1)`

error: called `filter(<f>).is_some()` on an `Option` value
  --> $DIR/option_filter_is_some.rs:25:17
   |
LL |     let _ = opt.filter(|x| x.eq(&5)).is_some();
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
help: use `is_some_and(<f>)` instead
   |
LL -     let _ = opt.filter(|x| x.eq(&5)).is_some();
LL +     let _ = opt.is_some_and(|x| (&x).eq(&5));
   |

error: called `filter(<f>).is_some()` on an `Option` value
  --> $DIR/option_filter_is_some.rs:27:17
   |
LL |     let _ = opt.filter(|&x| x > 1).is_some();
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
help: use `is_some_and(<f>)` instead
   |
LL -     let _ = opt.filter(|&x| x > 1).is_some();
LL +     let _ = opt.is_some_and(|x| x > 1);
   |

error: aborting due to 5 previous errors

